packages/cloud_functions/lib/cloud_functions.dart

```diff
@@ -43,10 +43,7 @@ class CloudFunctions {
       {@required String functionName, Map<String, dynamic> parameters}) async {
```

Comment:
I don't know what this is supposed to return, but you could add a similar feature to the invokeMethod to allow users to use inference instead of casting:

```
Future<T> call(	  Future<dynamic> call<T>(
      {@required String functionName, Map<String, dynamic> parameters}) async {
```

Where if `T` isn't specified it defaults to dynamic. Example:

```
final bool x = cloudFunctions.call(functionName: 'hello'); // <T> inferred as bool
final y = cloudFunctions.call<String>(functionName: 'bar'); // <T> specified as String
```

Reply:
SG, I think we can follow this up in a different PR. 
Also, by looking at the implementation, this method should return Future<Map<String, dynamic>>, however it would be breaking change since call is a public method. I would avoid that in this PR. And after this is merged and if we think it is necessary to break, then we can create another PR to fix this.
There are other places that has public API returning <dynamic> which I am also avoiding in this PR.